### PR TITLE
Remove deprecated configs from tests and documentation

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

DATADOG_TRACE_DEBUG=true does not follow the convention of the other tracers (.NET and Java) which use DD_TRACE_DEBUG=true . Apparently the Python tracer can take both configs, but we should change this in the documentation.

## Checklist
- [x] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (https://github.com/DataDog/documentation/pull/11152).

### CODE REVIEW

File: tests/commands/test_runner.py [hunk 1]
```diff
@@ -86,16 +86,16 @@ def test_integration(self):
 
     def test_debug_enabling(self):
         """
-        {DD,DATADOG}_TRACE_DEBUG=true allows setting debug logging of the global tracer
+        DD_TRACE_DEBUG=true allows setting debug logging of the global tracer.
         """
-        with self.override_env(dict(DATADOG_TRACE_DEBUG="false")):
+        with self.override_env(dict(DD_TRACE_DEBUG="false")):
```
Comment (brettlangdon) [inline tests/commands/test_runner.py:91]: we might want to keep a test case with `DATADOG_TRACE_DEBUG` for backwards compatibility checking

File: tests/commands/test_runner.py [hunk 2]
```diff
@@ -86,16 +86,16 @@ def test_integration(self):
 
     def test_debug_enabling(self):
         """
-        {DD,DATADOG}_TRACE_DEBUG=true allows setting debug logging of the global tracer
+        DD_TRACE_DEBUG=true allows setting debug logging of the global tracer.
         """
-        with self.override_env(dict(DATADOG_TRACE_DEBUG="false")):
```
Comment (Kyle-Verhoog) [inline tests/commands/test_runner.py:91]: Probably worth keeping some tests with the old variable around to ensure we don't break backwards compatibility

**Review by Kyle-Verhoog — COMMENTED** [review]
Just one concern about the tests but otherwise lgtm! :+1:

**Review by brettlangdon — COMMENTED** [review]
we also want to update our corp docs as well: https://github.com/DataDog/documentation/search?q=DATADOG_TRACE_DEBUG

File: tests/commands/test_runner.py (hunk 1 above)
Comment (mabdinur) [reply tests/commands/test_runner.py:91]: > we also want to update our corp docs as well: https://github.com/DataDog/documentation/search?q=DATADOG_TRACE_DEBUG

I opened this pr: https://github.com/DataDog/documentation/pull/11152

**Review by Kyle-Verhoog — DISMISSED** [review]

**Review by brettlangdon — DISMISSED** [review]

**Review by P403n1x87 — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**